Flaky test TestSchedulerShutdown_FrontendLoop

**Describe the bug**

From https://github.com/grafana/mimir/runs/3444635023?check_suite_focus=true

```
 ok  	github.com/grafana/mimir/pkg/ruler/rulestore/objectclient	0.052s [no tests to run]
--- FAIL: TestSchedulerShutdown_FrontendLoop (0.01s)
    scheduler_test.go:286: 
        	Error Trace:	scheduler_test.go:286
        	Error:      	Should be true
        	Test:       	TestSchedulerShutdown_FrontendLoop
FAIL
FAIL	github.com/grafana/mimir/pkg/scheduler	2.172s
```

**To Reproduce**

TBD. Test passed on a rerun.

**Expected behavior**

Tests to pass since the PR in question didn't involve this test or code.

**Environment:**
 - Infrastructure: [e.g., Kubernetes, bare-metal, laptop]
 - Deployment tool: [e.g., helm, jsonnet]

**Storage Engine**
- [x] Blocks
- [ ] Chunks

--- a/pkg/scheduler/scheduler_test.go
+++ b/pkg/scheduler/scheduler_test.go
@@ -271,7 +271,7 @@ func TestSchedulerShutdown_FrontendLoop(t *testing.T) {
 	frontendLoop := initFrontendLoop(t, frontendClient, "frontend-12345")
 
 	// Stop the scheduler. This will disable receiving new requests from frontends.
-	scheduler.StopAsync()
+	require.NoError(t, services.StopAndAwaitTerminated(context.Background(), scheduler))
 
 	// We can still send request to scheduler, but we get shutdown error back.
 	require.NoError(t, frontendLoop.Send(&schedulerpb.FrontendToScheduler{
@@ -307,7 +307,7 @@ func TestSchedulerShutdown_QuerierLoop(t *testing.T) {
 	_, err = querierLoop.Recv()
 	require.NoError(t, err)
 
-	scheduler.StopAsync()
+	require.NoError(t, services.StopAndAwaitTerminated(context.Background(), scheduler))
 
 	// Unblock scheduler loop, to find next request.
 	err = querierLoop.Send(&schedulerpb.QuerierToScheduler{})

